pkg/testcase/cluster.go

```diff
@@ -12,32 +12,28 @@ import (
 )
 
 func TestBuildCluster(g GinkgoTInterface) {
-	var err error
-
-	cluster := factory.GetCluster(g)
+	cluster := factory.AddCluster(g)
 	Expect(cluster.Status).To(Equal("cluster created"))
 	Expect(shared.KubeConfigFile).ShouldNot(BeEmpty())
 	Expect(cluster.ServerIPs).ShouldNot(BeEmpty())
 
-	if cluster.ProductType == "k3s" {
-		if strings.Contains(cluster.ClusterType, "etcd") {
-			fmt.Println("Backend:", cluster.ClusterType)
-		} else {
-			fmt.Println("Backend:", cluster.ExternalDb)
-		}
-	
-		if cluster.ExternalDb != "" && cluster.ClusterType == "" {
-			for i := 0; i > len(cluster.ServerIPs); i++ {
-				cmd := "grep \"datastore-endpoint\" /etc/systemd/system/k3s.service"
-				res, err := shared.RunCommandOnNode(cmd, cluster.ServerIPs[0])
-				Expect(err).NotTo(HaveOccurred())
-				Expect(res).Should(ContainSubstring(cluster.RenderedTemplate))
-			}
+	if strings.Contains(cluster.Config.DataStore, "etcd") {
```

Comment:
Will this work for k3s?

Reply:
we are alreading doing this validation earlier 

```
func addClusterConfig(
	g GinkgoTInterface,
	varDir string,
	terraformOptions *terraform.Options,
) (*Cluster, error) {
	...

	if cfg.Product == "k3s" {
		c.Config.DataStore = terraform.GetVariableAsStringFromVarFile(g, varDir, "datastore_type")
		if c.Config.DataStore == "" {
			c.Config.ExternalDb = terraform.GetVariableAsStringFromVarFile(g, varDir, "external_db")
			c.Config.RenderedTemplate = terraform.Output(g, terraformOptions, "rendered_template")
		}
	}
... 
```

Reply:
Sorry I missed that